components/portal/messages/controllers.js

```diff
@@ -66,27 +66,36 @@ define(['angular'], function(angular) {
          */
         var filterMessages = function() {
           // Check if group filtering has been disabled
-          if (!$localStorage.disableGroupFilteringForMessages) {
+          if ($localStorage.disableGroupFilteringForMessages) {
             // Define promises to run if filtering is turned on
             promiseFilteredMessages = {
               filteredByGroup:
+                messagesService.getAllMessages(),
```

Comment:
Not sure I like repeating this call that happens a few lines above -- we could just use an automatic resolve to `allMessages` so avoid another $http request